src/lexer.c

```diff
@@ -403,14 +471,34 @@ void tryParseExportStatement () {
 
 char16_t readExportAs (char16_t* startPos, char16_t* endPos) {
   char16_t ch = *pos;
+
   if (ch == 'a') {
     pos += 2;
     ch = commentWhitespace(true);
-    startPos = pos;
-    readToWsOrPunctuator(ch);
-    endPos = pos;
+
+    if (!isQuote(ch)) {
+      startPos = pos;
+      ch = readToWsOrPunctuator(ch);
+      endPos = pos;
+    }
+    // export { mod as "identifer" } from
+    // export { mod as "@notid" } from
+    // export { mod as "spa ce" } from
+    // export { mod as " space" } from
+    // export { mod as "space " } from
+    // export { mod as "not~id" } from
+    // export { mod as "%notid" } from
+    else {
+      startPos = pos + 1;
+
+      stringLiteral(ch);
+      readToWsOrPunctuator(ch);
```

Comment:
This line / call isn't needed - just string literal gets us to the right position.